### ISSUE TITLE
Quiz allows incorrect answer to pass

### ISSUE THREAD

**Issue by SherylHohman** [udacity/dlnd-issue-reports#494]
https://classroom.udacity.com/nanodegrees/nd101/parts/73a7fe8e-4354-4362-939d-a8bf2bae870d/modules/e6a4e4a1-98a9-4afe-a744-7ae6a59c01b8/lessons/0e07fafa-e796-4fab-b119-13f47f1d5c1b/concepts/b2bbdc9a-9f48-4735-b408-71cf67f5b000

In my implementationof this quiz, I never divided by the number of samples, yet my answer passed.

I went on to use my code from this quiz as a reference for Project 1.
Project 1 pased the Unit Tests (which also do not make this check.).
I went on to train my network, and set good parameters.  My model was Great.
But when I went to submit, pa said my backprop was wrong, and my weights_update was wrong (and of course my lr was wrong as well).

Eventually I submitted anyway, after urging from several others to do so.  My project passed as this was my only error.  My Learning Rate param appropriately compensated for not dividing my delta weights updater by 128, the sample size.  
I essentially obtained the same results, but it was not technically the correct/best/expected solution.

My reviewer caught the discrepancy and passed my project, providing me with an explanation (that was a good learning moment - very well handled

I applaud my reviewer passing the project despite this "small" error, after all the distress I had by the contradictory errors (unit tests passed).  I was at such a loss, I eventually took peoples advice to submit anyway.
I was relieved when I saw my project had passed - it was a positive feedback that my intuition to what a well trained network looked like was correct, and that I "mostly" understood (at a basic level) how backprop worked, and could implement a basic rendition of it.

I did go back to my project to correct this error, even though it could not be resubmitted.

I had suggested that we should be quizzed on this point, especially since it has such large implications for the success of our project.  
Someone pointed out to me that this lecture *did just that*.

But my implementation of this exercise missed that point altogether.  I referenced my code for this quiz throughout the remainder of the Lesson 2 lectures, and while working on my project.

Neither this quiz, nor the Unit Tests for Project 1 test for dividing by the number of samples in a batch.  As such, if we missed this detail, we are left in the dark as to what we are doing wrong.  And given matrix multiplication that can go awry, we are given to think that our entire understanding of the Big Concepts in backprop are completely incorrect.  We do not have a base to find or figure out that *this* is our problem.

Please see if there is a way to update this quiz to verify that we have taken the length of the dataset into consideration.
There is No other exercise that even touches on this.

Thank you for considering this change.
I believe it will be helpful to future students.

Other parts of the lecture material do mention "averages" but mostly it is glossed over, and /or gets "absorbed" in a constant.

This is the *only* location (that I'm aware of) that has us Code this detail.
And we don't even *have* to Code it in this lesson.  Case in point - I did not, 
I Never once coded it until after I completed "My First Neural Network".  And only then, due to FANTASTIC Guidance by my Reviewer.

I'd have never found this error on my own.

**Comment by Ortalarel** [comment]
thank you for the feedback. We are looking into it.
